compiler/dfa.nim

```diff
@@ -609,15 +609,16 @@ proc aliases*(obj, field: PNode): bool =
 
 proc useInstrTargets*(ins: Instr; loc: PNode): bool =
   assert ins.kind == use
-  sameTrees(ins.n, loc) or
-  ins.n.aliases(loc) or loc.aliases(ins.n) # We can come here if loc is 'x.f' and ins.n is 'x' or the other way round.
+  result = sameTrees(ins.n, loc) or
+    ins.n.aliases(loc) or loc.aliases(ins.n)
+  # We can come here if loc is 'x.f' and ins.n is 'x' or the other way round.
   # use x.f;  question: does it affect the full 'x'? No.
   # use x; question does it affect 'x.f'? Yes.
 
 proc defInstrTargets*(ins: Instr; loc: PNode): bool =
   assert ins.kind == def
-  sameTrees(ins.n, loc) or
-  ins.n.aliases(loc) # We can come here if loc is 'x.f' and ins.n is 'x' or the other way round.
+  result = sameTrees(ins.n, loc) or ins.n.aliases(loc)
```

Comment:
~~Weird, when I tried to fix this, this didn't work.. ~~ 
EDIT: Nevermimd, missed the `gen` call.
Now use/defInstrTargets can be one and the same \o/